hypothesis-python/src/hypothesis/extra/_array_helpers.py

```diff
@@ -632,12 +632,15 @@ def _draw_loop_dimensions(self, data, use=None):
 
 
 class BasicIndexStrategy(st.SearchStrategy):
-    def __init__(self, shape, min_dims, max_dims, allow_ellipsis, allow_newaxis):
+    def __init__(
+        self, shape, min_dims, max_dims, allow_ellipsis, allow_newaxis, flat_index
```

Comment:
I think a more verbose argument name might be in order, since it's internal-only.  How about `allow_fewer_indices_than_dims`?

Then the new comment can move up here too, and probably be somewhat shorter - e.g. "The Array API spec requires that indexers either contain ellipsis, or contain an index for every array dimension".